x-pack/plugins/monitoring/public/application/contexts/global_state_context.tsx

```diff
@@ -42,9 +43,11 @@ export const GlobalStateProvider: React.FC<GlobalStateProviderProps> = ({
   children,
 }) => {
   const localState: State = {};
-  const state = new GlobalState(query, toasts, localState as { [key: string]: unknown });
+  const [globalState] = useState(
+    new GlobalState(query, toasts, localState as { [key: string]: unknown })
```

Comment:
I think if the Provider starts to rerender multiple times, then we will still have a bug here as a new instance will be created on each render and internally, there will be subscribers that will never unsubscribe. 

I think we could fix this by wrapping into factory function: 

 ```suggestion
    () => new GlobalState(query, toasts, localState as { [key: string]: unknown })
```

Reply:
Apologies, this was already supposed to be inside a lazy initialisation function. Don't fix bugs on Mondays 🙈